jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiException.java

```diff
@@ -24,195 +24,130 @@
  *
  * The MultiException itself should not be thrown instead one of the
  * ifExceptionThrow* methods should be called instead.
+ * TODO: Remove entirely in favor of Error lists with {@link TypeUtil#withSuppressed(Throwable, List)}?
+ * TODO: Deprecate this now (if we do above)?
  */
 @SuppressWarnings("serial")
 public class MultiException extends Exception
 {
     private static final String DEFAULT_MESSAGE = "Multiple exceptions";
 
-    private List<Throwable> nested;
-
     public MultiException()
     {
-        // Avoid filling in stack trace information.
-        super(DEFAULT_MESSAGE, null, false, false);
-        this.nested = new ArrayList<>();
+        this(DEFAULT_MESSAGE);
     }
 
-    /**
-     * Create a MultiException which may be thrown.
-     *
-     * @param nested The nested exceptions which will be suppressed by this
-     * exception.
-     */
-    private MultiException(List<Throwable> nested)
+    public MultiException(String message)
     {
-        super(DEFAULT_MESSAGE);
-        this.nested = new ArrayList<>(nested);
-
-        if (nested.size() > 0)
-            initCause(nested.get(0));
-
-        for (Throwable t : nested)
-        {
-            if (t != this && t != getCause())
-                addSuppressed(t);
-        }
+        // Avoid filling in stack trace information.
+        super(message, null, true, false);
     }
 
     public void add(Throwable e)
     {
-        if (e instanceof MultiException)
+        if (e instanceof MultiException mex)
```

Comment:
Before adding suppress we should check org.eclipse.jetty.util.TypeUtil#isAssociated.   Perhaps this entire method can just be:
```suggestion
        if (!TypeUtil#isAssociated(this,e))
            addSuppressed(e);
```